soh/soh/Enhancements/randomizer/randomizer_check_tracker.cpp

```diff
@@ -787,9 +822,9 @@ void Teardown() {
 
 void UpdateCheck(uint32_t check, RandomizerCheckTrackerData data) {
     auto area = RandomizerCheckObjects::GetAllRCObjects().find(static_cast<RandomizerCheck>(check))->second.rcArea;
-    if (!gSaveContext.checkTrackerData[check].skipped && data.skipped) {
+    if ((!gSaveContext.checkTrackerData[check].skipped && data.skipped) || (gSaveContext.checkTrackerData[check].status >= RCSHOW_COLLECTED && data.status < RCSHOW_COLLECTED)) {
         areaChecksGotten[area]++;
-    } else if (gSaveContext.checkTrackerData[check].skipped && !data.skipped) {
+    } else if ((gSaveContext.checkTrackerData[check].skipped && !data.skipped) || (gSaveContext.checkTrackerData[check].status < RCSHOW_COLLECTED && data.status >= RCSHOW_COLLECTED)) {
```

Comment:
it's not really clear what `< RCSHOW_COLLECTED` and `>= RCSHOW_COLLECTED` means here. i think even if it's more verbose it would make sense to have this be

```suggestion
    } else if ((gSaveContext.checkTrackerData[check].skipped && !data.skipped) || ((gSaveContext.checkTrackerData[check].status != RCSHOW_COLLECTED && gSaveContext.checkTrackerData[check].status != RCSHOW_SAVED) && (data.status == RCSHOW_COLLECTED || data.status == RCSHOW_SAVED))) {
```

Reply:
Using >= or < here was just an attempt to shorten the code necessary. You're right that >= would mean it was either collected or saved and < is anything else. I don't particularly care either way, I just like more concise code that does the same job.

Reply:
when reading `>=` i had no idea what it meant and had to go look up the enum to understand the line. so while it may be fewer characters it too much longer to understand

Reply:
To be fair, though, it wasn't an || for the comparison for == COLLECTED or SAVED, but an &&. And, I don't use this myself, Anchor was what kept this in there, and I don't have any way of testing it to make sure it's actually working like I expect XD.

Reply:
> it wasn't an || for the comparison for == COLLECTED or SAVED, but an &&

i'm not sure what you mean here

```cpp
gSaveContext.checkTrackerData[check].status < RCSHOW_COLLECTED
```
is logically equivalent to
```cpp
gSaveContext.checkTrackerData[check].status != RCSHOW_COLLECTED && gSaveContext.checkTrackerData[check].status != RCSHOW_SAVED
```
and
```cpp
data.status >= RCSHOW_COLLECTED
```
is logically equivalent to
```cpp
data.status == RCSHOW_COLLECTED || data.status == RCSHOW_SAVED
```

Reply:
Oh, I misread your suggestion. I thought you were using || in the first instance for some reason.